processing/src/main/java/org/apache/druid/segment/DimensionHandlerUtils.java

```diff
@@ -61,6 +61,8 @@ private DimensionHandlerUtils()
   public static final ColumnCapabilities DEFAULT_STRING_CAPABILITIES =
       new ColumnCapabilitiesImpl().setType(ValueType.STRING)
                                   .setDictionaryEncoded(true)
+                                  .setDictionaryValuesUnique(true)
```

Comment:
This doesn't seem right. The only place `DEFAULT_STRING_CAPABILITIES` is `getEffectiveCapabilities` in this same file. And looking at the cases in that method, each one seems like it should behave a bit differently. So consider deleting this constant and having the `getEffectiveCapabilities` should be generating various capabilities objects.